snapcraft/plugins/kernel.py

```diff
@@ -142,16 +141,14 @@ def __init__(self, name, options):
         self.make_install_targets.extend(self._get_fw_install_targets())
 
         self.os_snap = os.path.join(self.sourcedir, 'os.snap')
-        self._target_arch = get_machine_info(target_machine)
 
-    def set_target_machine(self, machine):
-        self._target_arch = get_machine_info(machine)
+    def set_target_machine(self):
```

Comment:
I thought about this, but if we make plugins take the charge on this I can add something to manage this. The problem is plugins that don't support cross compiling.